components/ui/carousel.tsx

```diff
@@ -136,6 +138,14 @@ const Carousel = React.forwardRef<
       };
     }, [api, onSelect]);
 
+    React.useEffect(() => {
+      if (!api || !itemLength) {
+        return;
+      }
+      api?.scrollTo(Math.floor((itemLength - 1) / 2));
+      setSelectedIndex(Math.floor((itemLength - 1) / 2));
+    }, [itemLength, api]);
+
```

Comment:
returning inside of a useEffect doesn't exit the function (assuming that's what you're trying to do). It's used for adding cleanup code. See [docs](https://react.dev/reference/react/useEffect#connecting-to-an-external-system)

```suggestion

    React.useEffect(() => {
    
      const lengthAndApiMissing = !api || !itemLength
      if (!lengthAndApiMissing ) {
           api?.scrollTo(Math.floor((itemLength - 1) / 2));
           setSelectedIndex(Math.floor((itemLength - 1) / 2));
      }
    }, [itemLength, api]);

```

Reply:
Forgot about that one – good catch